eva-tools/src/test/java/embl/ebi/variation/eva/vcfDump/VariantExporterControllerTest.java

```diff
@@ -0,0 +1,331 @@
+import org.opencb.opencga.lib.common.Config;
+import org.opencb.opencga.storage.core.StorageManagerException;
+import org.opencb.opencga.storage.core.StorageManagerFactory;
+import org.opencb.opencga.storage.core.variant.VariantStorageManager;
+import org.opencb.opencga.storage.core.variant.adaptors.VariantDBAdaptor;
+import org.opencb.opencga.storage.core.variant.adaptors.VariantDBIterator;
+import org.opencb.opencga.storage.core.variant.adaptors.VariantSourceDBAdaptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by pagarcia on 10/05/2016.
+ */
+public class VariantExporterControllerTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private static final String DB_NAME = "VariantExporterTest";
+    public static final String OUTPUT_DIR = "/tmp/";
+
+    private static CellbaseWSClient cellBaseClient;
+    private static VariantStorageManager variantStorageManager;
+    private static VariantDBAdaptor variantDBAdaptor;
+    private static VariantSourceDBAdaptor variantSourceDBAdaptor;
+    private static final Logger logger = LoggerFactory.getLogger(VariantExporterControllerTest.class);
+    private static final MultivaluedMap<String, String> emptyFilter = new MultivaluedHashMap<>();
+
+
+    @BeforeClass
+    public static void setUpClass() throws IllegalAccessException, ClassNotFoundException, InstantiationException, URISyntaxException, StorageManagerException, IOException, InterruptedException {
+        VariantExporterTestDB.cleanDBs();
+        VariantExporterTestDB.fillDB();
+
+        Config.setOpenCGAHome(System.getenv("OPENCGA_HOME") != null ? System.getenv("OPENCGA_HOME") : "/opt/opencga");
+        cellBaseClient = new CellbaseWSClient("hsapiens");
+        logger.info("Using cellbase: " + cellBaseClient.getUrl() + " version " + cellBaseClient.getVersion());
+        variantStorageManager = StorageManagerFactory.getVariantStorageManager();
+        variantDBAdaptor = variantStorageManager.getDBAdaptor(DB_NAME, null);
+        variantSourceDBAdaptor = variantDBAdaptor.getVariantSourceDBAdaptor();
+    }
+
+    /**
+     * Clears and populates the Mongo collection used during the tests.
+     *
+     * @throws UnknownHostException
+     */
+    @AfterClass
+    public static void tearDownClass() throws UnknownHostException {
+        VariantExporterTestDB.cleanDBs();
+    }
+
+    @Test
+    public void testGetQuery() throws Exception {
+        List<String> studies = Arrays.asList("s1", "s2");
+        List<String> files = Arrays.asList("f3", "f4", "f5");
+
+        VariantExporterController controller = new VariantExporterController("hsapiens", null, studies, files, null, emptyFilter);
```

Comment:
The constructor should check for parameters correctness, for instance, in a real use case it doesn't make sense that the database name is null.